grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/bazel/rules/MavenRules.kt

```diff
@@ -126,17 +126,23 @@ fun StatementsBuilder.mavenInstall(
                 }
             }
         }
+
+        val prefix = if (!mavenInstallJsonEnabled) "#" else ""
         mavenInstallJson?.let {
-            "maven_install_json" `=` mavenInstallJson.quote
+            "${prefix}maven_install_json" `=` "//:$mavenInstallJson".quote
         }
 
         versionConflictPolicy?.let {
             "version_conflict_policy" `=` it.quote
         }
+
+        if (artifactPinning) {
+            "fail_if_repin_required" `=` "False"
```

Comment:
is this going to potentially cause undesired outcome? since we might not be aware that a repin might be required?

Reply:
Good point, in that case a warning would be printed to console. The idea is same as current, i.e run `migrateToBazel` to update scripts including maven_install.json. If dependency is updated, running build without `migrateToBazel` is undefined behavior (same as current without pinning).  Making this true leads to extra cases handling in buildifier download etc.